### PR TITLE
[Storage][DataMovement] Refactor download transfers

### PR DESCRIPTION
**Pull request by jalauzon-msft**
This change refactors download transfers to hopefully improve performance and simplify the code:
- Remove the eventing for `DownloadChunkHandler` as it wasn't needed. The job part now simply adds directly to the channel.
- Removed "early bird" chunks entirely. Chunks were written in order to disk and chunks that came out of order were saved to a temporary file. Now chunks are written in whatever order they come in by seeking to the proper location in the file for each chunk.
  - Further testing needs to be done to see if this is performant enough. Further improvements could also be made like somehow leaving the file stream open for the duration of that job part transfer, so we don't continuously open and close the file.
- Other minor refactoring.

### CODE REVIEW

**Comment by jalauzon-msft** [comment]
Perf results - https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4345659&view=logs&j=42ded549-05ee-5cdb-7ba7-7a948a0cc056&t=1a18f1cb-d016-5e13-bec4-a5dc2ac35679

**Review by jaschrep-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.DataMovement/src/DownloadChunkHandler.cs [hunk 1]
```diff
@@ -2,80 +2,47 @@
...
-        /// Create channel of <see cref="DownloadRangeEventArgs"/> to keep track of that are
-        /// waiting to update the bytesTransferred and other required operations.
+        /// Create channel of <see cref="QueueDownloadChunkArgs"/> to keep track to handle
+        /// writing downloaded chunks to the destination as well as tracking overall progress.
         /// </summary>
-        private readonly Channel<DownloadRangeEventArgs> _downloadRangeChannel;
-        private CancellationToken _cancellationToken;
+        private readonly Channel<QueueDownloadChunkArgs> _downloadRangeChannel;
+        private readonly Task _processDownloadRangeEvents;
+        private readonly CancellationToken _cancellationToken;
```
[the remark this answers is not in the document]
Comment (jalauzon-msft) [reply sdk/storage/Azure.Storage.DataMovement/src/DownloadChunkHandler.cs:41]: Will take this on in the follow-up PR to add bounds to some of our channels. That will require some changes to the `ChannelProcessor` and the channel in this class requires some careful considerations to writing and completing the channel that we have not dealt with yet with the existing channels.

**Pull request merged**